### DISCUSSION TITLE
Problems with combining carla and omnisafe

### DISCUSSION THREAD

**Discussion by lyy2109** [PKU-Alignment/omnisafe#307] [Q&A]
Can I customize the vehicle simulation environment based on carla_gym in omnisafe?

**Comment by Gaiejj** [comment]
Same issue here: #305

**Reply by lyy2109** [reply]
Thank you for your answer. In fact, I don’t quite understand the meaning of [#305]. I plan to use CPO to train a simulated vehicle environment created by Carla-gym, and do not plan to use multi-agent reinforcement learning algorithms or parallel environments (vector environments). I would like to ask if this idea is feasible?

**Reply by Gaiejj** [reply]
I apologize for my misunderstanding. OmniSafe supports operation in various customized environments. You only need to create a new file under `omnisafe/envs` for building the environment from carla-gym, such as `carla_gym.py`. Afterwards, follow the input and output format of the methods in `safety_gymnasium.py` file to implement the corresponding functions. For details, please refer to our README and #285.

**Reply by lyy2109** [reply]
Thanks for your reply. I will give it a try and will ask again if any problem arises.
